x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -564,45 +474,42 @@ export class Authenticator {
     }
   }
 
-  /**
-   * Extracts session value for the specified request. Under the hood it can
-   * clear session if it belongs to the provider that is not available.
-   * @param sessionStorage Session storage instance.
-   */
-  private async getSessionValue(sessionStorage: SessionStorage<ProviderSession>) {
-    const sessionValue = await sessionStorage.get();
-
-    // If we detect that session is in incompatible format or for some reason we have a session
-    // stored for the provider that is not available anymore (e.g. when user was logged in with one
-    // provider, but then configuration has changed and that provider is no longer available), then
-    // we should clear session entirely.
-    if (
-      sessionValue &&
-      (isLegacyProviderSession(sessionValue) ||
-        this.providers.get(sessionValue.provider.name)?.type !== sessionValue.provider.type)
-    ) {
-      sessionStorage.clear();
-      return null;
-    }
-
-    return sessionValue;
-  }
-
-  private updateSessionValue(
-    sessionStorage: SessionStorage<ProviderSession>,
+  private async updateSessionValue(
```

Comment:
__note:__ this method is a "bit" complex and hard to follow, let me know if you have any questions or improvement proposals!